test/e2e/flask/multichain-api/get-session.spec.ts

```diff
@@ -0,0 +1,146 @@
+        }) => {
+          await openDappAndConnectWallet(driver, extensionId);
+
+          await driver.clickElement('#get-session-btn');
+
+          const notificationList = await driver.findElements(
+            '.notification-container',
+          );
+          const sessionResultSummary = notificationList[0];
+          await sessionResultSummary.click();
+
+          const completeRawResult = await driver.findElement(
+            '#session-method-result-0',
+          );
+          const parsedResult = JSON.parse(await completeRawResult.getText());
+
+          if (!_.isEmpty(parsedResult.sessionScopes)) {
+            throw new Error('Should receive empty session scopes');
+          }
+        },
+      );
+    });
+  });
+
+  describe('Connect wallet to the multichain dapp via `externally_connectable`, call `wallet_getSession` with multiple sessions created:', function () {
+    it('should successfully receive result that specifies its permitted session scopes for selected chains', async function () {
+      await withFixtures(
+        {
+          title: this.test?.fullTitle(),
+          ...OPTIONS,
+        },
+        async ({
+          driver,
+          extensionId,
+        }: {
+          driver: Driver;
+          extensionId: string;
+        }) => {
+          await openDappAndConnectWallet(driver, extensionId);
+          const sessionScopes = ['eip155:1', 'eip155:59144', 'eip155:42161'];
+
+          for (const scope of sessionScopes) {
+            await driver.clickElement(`input[name="${scope}"]`);
+          }
+
+          await driver.clickElement('#create-session-btn');
+          await driver.switchToWindowWithTitle(WINDOW_TITLES.Dialog);
+          await driver.delay(largeDelayMs);
+
+          const editButtons = await driver.findElements('[data-testid="edit"]');
+          await editButtons[1].click();
+
+          await driver.delay(largeDelayMs);
+
+          await driver.clickElement(
+            '[data-testid="connect-more-chains-button"]',
+          );
+
+          await driver.clickElement({ text: 'Connect', tag: 'button' });
+
+          await driver.switchToWindowWithTitle(
+            WINDOW_TITLES.MultichainTestDApp,
+          );
+
+          await driver.clickElement('#get-session-btn');
+
+          const notificationList = await driver.findElements(
+            '.notification-container',
+          );
+
+          const sessionResultSummary = notificationList[0];
+          await sessionResultSummary.click();
+
+          const completeRawResult = await driver.findElement(
+            '#session-method-result-0',
+          );
+          const parsedResult = JSON.parse(await completeRawResult.getText());
+
+          for (const scope of sessionScopes) {
+            if (!parsedResult.sessionScopes[scope]) {
+              throw new Error(
+                `Should receive result that specifies session scopes for ${scope}`,
+              );
+            }
+          }
+        },
```

Comment:
thoughts on checking for full object equality? methods, notifications, accounts, everything? not just if the scope exists

Reply:
Yeah definitely!

Reply:
Resolved in [a105b35](https://github.com/MetaMask/metamask-extension/pull/29298/commits/a105b35b9d82ff0657fb79f9a42d29760b21e09a)

`getExpectedSessionScope` helper util function in `test/e2e/flask/multichain-api/testHelpers.ts` gets us the expected object to compare to.
I notice we do have the `methods` that can be present in a scope in this file `app/scripts/controllers/permissions/specifications.js`, but with a lot more of them.
I'm not 100% satisfied with repeating some of these hardcoded strings, but also don't think adding complexity by filtering the existing `unrestrictedMethods` and `unrestrictedEthSigningMethods` to only return the methods that we pass down to the scope.

So is there any place I might be missing in the code base where I can get this exact list of methods that I'm coding again inside `methods` in the helper function ? lmk 👍🏾

Reply:
https://github.com/MetaMask/core/blob/main/packages/multichain/src/scope/constants.ts is a good place to get these